client/settings/views/services/group.js

```diff
@@ -1,27 +1,25 @@
 import React, { PropTypes } from 'react';
 import ShippingServiceEntry from './entry';
 import FoldableCard from 'components/foldable-card';
-import CheckBox from 'components/forms/form-checkbox';
+import IndeterminateCheckbox from 'components/indeterminate-checkbox';
 import Gridicon from 'components/gridicon';
 import InfoTooltip from 'components/info-tooltip';
 import { translate as __ } from 'lib/mixins/i18n';
 import { sprintf } from 'sprintf-js';
 import _ from 'lodash';
 
-const summaryLabel = ( services ) => {
-	const numSelected = services.reduce( ( count, service ) => (
-		count + ( service.enabled ? 1 : 0 )
-	), 0 );
+const summaryLabel = ( services, numSelected ) => {
 	if ( numSelected === services.length ) {
 		return __( 'All services selected' );
 	}
 	const format = ( 1 === numSelected ) ? __( '%d service selected' ) : __( '%d services selected' );
 	return sprintf( format, numSelected );
 };
 
-const updateAll = ( event, updateValue, services ) => {
+const updateAll = ( event, checked, updateValue, services ) => {
+	event.stopPropagation();
 	services.forEach( ( service ) => {
-		updateValue( [ service.id, 'enabled' ], event.target.checked );
+		updateValue( [ service.id, 'enabled' ], checked );
```

Comment:
Same here. The reducer has enough information to know if a toggle will turn on/off all the services, this extra param is redundant.